File: Study Plans/Binary Search I/Day 8/74. Search a 2D Matrix.py
class Solution:
    def searchMatrix(self, matrix, target):
        if not matrix or target is None:
            return False
        row, col = 0, len(matrix[0])-1
        while col >= 0 and row <= len(matrix)-1:
            if matrix[row][col] == target:
                return True
            elif matrix[row][col] > target:
                col = col-1
            else:
                row = row + 1
        return False

File: Study Plans/Binary Search I/Day 8/74. Search a 2D Matrix.cpp
// Question 2

// Problem Link
// https://leetcode.com/problems/search-a-2d-matrix/

// Solution:
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int n = matrix.size(), m = matrix[0].size(), id = 0;
        for(int i = 0; i < matrix.size(); i++) {
            if(matrix[i][0] == target || matrix[i][m-1] == target)
                return true;
            else if(matrix[i][m-1] > target) {
                id = i;
                break;
            }
        }
        if(matrix[id][0] > target)
            return false;
        for(int j = 0; j < m; j++)
            if(matrix[id][j] == target)
                return true;
        return false;
    }
};